### ISSUE TITLE
Enable flash attention

### ISSUE THREAD

**Issue by inatatsu** [torch-spyre/torch-spyre#991]
### What feature would you like to be added?

Enable [Flash Attention](https://arxiv.org/abs/2205.14135) for Spyre, which calculates an online softmax distribution of a partial scaled dot product on the LX memory.

### Why is this feature needed?

This operator is used by important models:

- torch-spyre/sendnn-inference#696 
- torch-spyre/sendnn-inference#698 

### Any alternative solution that you may want to share?

None.

### Anything else we need to know?

**PyTorch/FlexAttention**
- [`scaled_dot_product_attention_for_cpu`](https://github.com/pytorch/pytorch/blob/v2.10.0/torch/_decomp/decompositions.py#L5104-L5182)
- [FlexAttention: The Flexibility of PyTorch with the Performance of FlashAttention](https://pytorch.org/blog/flexattention/)
- [`torch.nn.attention.flex_attention`](https://docs.pytorch.org/docs/stable/nn.attention.flex_attention.html) 

**MTIA**
- [Four MTIA Chips in Two Years: Scaling AI Experiences for Billions](https://ai.meta.com/blog/meta-mtia-scale-ai-chips-for-billions/)

**TPU**
- [Custom Kernels via Pallas](https://docs.pytorch.org/xla/master/features/pallas.html)

**Gaudi**
- [Using Fused Scaled Dot Product Attention (FusedSDPA)](https://docs.habana.ai/en/latest/PyTorch/Model_Optimization_PyTorch/Optimization_in_PyTorch_Models.html#using-fused-scaled-dot-product-attention-fusedsdpa)

**vLLM Spyre**
- https://github.com/vllm-project/vllm-spyre/issues/639
- https://github.com/vllm-project/vllm-spyre/issues/650

**Comment by inatatsu** [comment]
@moriohara Please assign this issue to me.

**Comment by ani300** [comment]
This is now working, in a reduced form, we can track improvements in follow-up issues
